### PR TITLE
El-MAVEN-Polly Integration

### PR DESCRIPTION
**Pull request by shubhra-agrawal**
Requirements:
- Add New Project option should be selected by default every time it opens
- Loading gif while projects are being fetched on opening polly interface anytime after the login
- Advanced settings show up in another dialog
- Go To Polly button takes user to the first view app of the last uploaded project

### CODE REVIEW

File: src/cli/peakdetector/PeakDetectorCLI.cpp [hunk 1]
```diff
@@ -677,55 +696,124 @@ void PeakDetectorCLI::saveJson(string setName) {
 		#endif
 
 		jsonReports=new JSONReports(mavenParameters);
-		jsonReports->saveMzEICJson(mavenParameters->outputdir + setName + ".json",
+		string fileName = mavenParameters->outputdir + setName + ".json";
+		if (uploadToPolly_bool){
+			string fileName = setName;
```
Comment (sahil21) [inline src/cli/peakdetector/PeakDetectorCLI.cpp]: @shubhra-agrawal @Ky5595 fileName scope will end after the if statement. You should not initialize it again.

File: src/gui/mzroll/ligandwidget.cpp [hunk 2]
```diff
@@ -514,6 +514,67 @@ void LigandWidget::saveCompoundList(){
     }
 }
 
+void LigandWidget::saveCompoundListToPolly(QString fileName,QString dbname){
```
Comment (sahil21) [inline src/gui/mzroll/ligandwidget.cpp]: @Ky5595 @shubhra-agrawal Why are we doing this? Why can't just take the file from user's system?

Comment (Ky5595) [reply src/gui/mzroll/ligandwidget.cpp]: @sahil21  Basically, we want to upload the compound database that was used to generate the peak table, to Polly. We could have had a mapping of the compound databases present in the ligand widget and their actual location on the system.
But, To validate it before uploading, we had to write a function separately. Personally, I feel that will add one more redundant function to the code, so i used already existing functions from Elmaven code.

This takes care of the following scenarios -
1. If the user uploads some compound dataset to Elmaven, does some analysis and for some reason that file is deleted from the system.
If we use fixed mapping, Elmaven will crash. 
2. If the file is modified on the system, Elmaven will upload the file but further workflow on Polly will break.

The current code takes care of both these scenarios by writing to a temporary file from RAM (Elmaven session variables) and then uploading it to Polly.

Comment (sahil21) [reply src/gui/mzroll/ligandwidget.cpp]: @Ky5595 saveCompoundList & saveCompoundListToPolly is sharing around 95% of code. Please refactor this.

File: src/gui/mzroll/tabledockwidget.cpp [hunk 3]
```diff
@@ -795,29 +795,75 @@ void TableDockWidget::exportGroupsToSpreadsheet() {
     }
 }
 
-void TableDockWidget::exportGroupsToSpreadsheet_polly(QString fileName) {
-    
+void TableDockWidget::prepareDataForPolly(QString writable_temp_dir,QString export_format,QString user_filename) {
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/tabledockwidget.cpp]: @Ky5595 exportGroupstoSpreadSheet already exists in tabledockwidget. The code is almost the same. Lets refactor this before merging.

File: src/gui/mzroll/tabledockwidget.cpp [hunk 4]
```diff
@@ -838,6 +884,26 @@ void TableDockWidget::exportGroupsToSpreadsheet_polly(QString fileName) {
     }
 }
 
+void TableDockWidget::exportJsonToPolly(QString writable_temp_dir,QString jsonfileName) {
```
Comment (shubhra-agrawal) [inline src/gui/mzroll/tabledockwidget.cpp]: @Ky5595 Similar function exists

**Pull request closed without merging**